### PR TITLE
HW8

### PR DESCRIPTION
**Pull request by iarkov**
Created branch HW8 with files Functions.cpp, FunctionsHW8.cpp, FunctionsHW8.h

### CODE REVIEW

File: FunctionsHW8.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <iostream>
```
Comment (bohdanpc) [inline FunctionsHW8.h]: don't include headers inside of headers unless you need them there right away

File: FunctionsHW8.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
+#include "FunctionsHW8.h"
+
+int factorialRec(unsigned int count) {
+	return (count == 0) ? 1 : count * factorialRec(count - 1);
```
Comment (bohdanpc) [inline FunctionsHW8.cpp:5]: count == 1

File: FunctionsHW8.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+int factorialLoop(unsigned int count) {
+
+	int factorial = 1;
+
+	if (count == 0) {
+		return factorial;
+	}
+	else {
```
Comment (bohdanpc) [inline FunctionsHW8.cpp]: else is redundant as there's `early return` above

File: FunctionsHW8.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+int factorialRec(unsigned int count) {
+	return (count == 0) ? 1 : count * factorialRec(count - 1);
+}
+
+int factorialLoop(unsigned int count) {
+
+	int factorial = 1;
+
+	if (count == 0) {
```
Comment (bohdanpc) [inline FunctionsHW8.cpp:12]: coutn == 1

File: FunctionsHW8.cpp [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+void helpFunctionD(int count, int i = 1) {
+	if (i == count) {
+		std::cout << i << std::endl;
+	}
+	else {
+		std::cout << i << " ";
+		helpFunctionD(count, i + 1);
+	}
+}
```
Comment (bohdanpc) [inline FunctionsHW8.cpp]: can be easily implemented without this helper function with 2 parameters

File: FunctionsHW8.cpp (hunk 2 above)
Comment (iarkov) [reply FunctionsHW8.cpp:5]: And if we put 0 as a parameter of factorialRec?

File: FunctionsHW8.cpp (hunk 4 above)
Comment (iarkov) [reply FunctionsHW8.cpp:12]: and if factorial of 0?

File: FunctionsHW8.cpp (hunk 5 above)
Comment (iarkov) [reply FunctionsHW8.cpp]: Right. Modified in next commit.

**Review by bohdanpc — APPROVED** [review]

**Pull request merged**